From a91266ef087532e2332c75c4fd9244df66f30b64 Mon Sep 17 00:00:00 2001
From: Daiki Ueno <ueno@gnu.org>
Date: Fri, 18 Dec 2020 13:37:10 +0100
Subject: [PATCH] meson: Link trust/client modules explicitly to -ldl

This adds the -ldl link flag missing in the meson build, but present
in the autotools build.  Although the use-case is unlikely, this
allows those modules to be linked as a normal shared library to a
program.
---
 p11-kit/meson.build | 1 +
 trust/meson.build   | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/p11-kit/meson.build b/p11-kit/meson.build
index 7d57cd7..02147a9 100644
--- a/p11-kit/meson.build
+++ b/p11-kit/meson.build
@@ -92,6 +92,7 @@ if host_system != 'windows'
                 'client.c', 'client-init.c',
                 name_prefix: '',
                 include_directories: [configinc, commoninc],
+                dependencies: dlopen_deps,
                 link_args: p11_module_ldflags,
                 link_depends: [p11_module_symbol_map,
                                p11_module_symbol_def],
diff --git a/trust/meson.build b/trust/meson.build
index 482a3c1..d4a8e15 100644
--- a/trust/meson.build
+++ b/trust/meson.build
@@ -56,7 +56,7 @@ shared_module('p11-kit-trust',
               'module-init.c',
               name_prefix: '',
               c_args: p11_kit_trust_c_args,
-              dependencies: [asn_h_dep, libp11_library_dep] + libtasn1_deps,
+              dependencies: [asn_h_dep, libp11_library_dep] + dlopen_deps + libtasn1_deps,
               link_args: p11_module_ldflags,
               link_depends: [p11_module_symbol_map,
                              p11_module_symbol_def],
-- 
2.29.2

